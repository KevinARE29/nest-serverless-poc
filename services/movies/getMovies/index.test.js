/**
 * @group unit
 */

const eventData = require('./exampleData.json')
const { handler } = require('./index')

describe('getMovies', () => {
  it('should retrieve all movies', async () => {
    const result = await handler(eventData)
    const body = JSON.parse(result.body)

    expect(result.statusCode).toBe(200)
    expect(body).toMatchSnapshot({
      data: expect.arrayContaining([
        {
          uuid: expect.any(String),
          name: expect.any(String),
          synopsis: expect.any(String),
          duration: expect.any(Number),
          price: expect.any(Number),
          likes: expect.any(Number),
          isActive: expect.any(Boolean),
          poster: expect.any(String),
        },
      ]),
      pagination: expect.objectContaining({
        perPage: expect.any(Number),
        page: expect.any(Number),
        total: expect.any(Number),
        totalPages: expect.any(Number),
      }),
    })
  })

  it('should return a 400 error if the pagination params are not valid', async () => {
    const result = await handler({
      ...eventData,
      queryStringParameters: {
        page: '0',
      },
    })
    const body = JSON.parse(result.body)

    expect(result.statusCode).toBe(400)
    expect(body.message).toBe('page must be a positive integer')
  })
})
