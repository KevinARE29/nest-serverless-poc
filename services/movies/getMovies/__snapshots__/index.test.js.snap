// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMovies should retrieve all movies 1`] = `
Object {
  "data": ArrayContaining [
    Object {
      "duration": Any<Number>,
      "isActive": Any<Boolean>,
      "likes": Any<Number>,
      "name": Any<String>,
      "poster": Any<String>,
      "price": Any<Number>,
      "synopsis": Any<String>,
      "uuid": Any<String>,
    },
  ],
  "pagination": ObjectContaining {
    "page": Any<Number>,
    "perPage": Any<Number>,
    "total": Any<Number>,
    "totalPages": Any<Number>,
  },
}
`;
